chain/beacon/sync_manager.go

```diff
@@ -181,7 +179,11 @@ func (s *SyncManager) CheckPastBeacons(ctx context.Context, upTo uint64, cb func
 
 	var faultyBeacons []uint64
 	// notice that we do not validate the genesis round 0
-	for i := uint64(1); i < uint64(s.store.Len()); i++ {
+	storeLen, err := s.store.Len(ctx)
```

Comment:
we identified elsewhere that this `Len` call is suuuuuper expensive and I actually raised a PR [here](https://github.com/drand/drand/pull/1085) to avoid using it elsewhere. I dunno if there's a way to get around it here but just a point to note

Reply:
getting the last beacon and using its round number + 1 is a filthy hack that works now, but won't work if we implemented DB trimming ;;

Reply:
For the Postgres storage, this would be a really fast call, but probably not many will run with it.
I wonder if we can keep track of the size in a different manner.
As for the DB trimming, that's likely going to cause issues, agreed.
Should we open a followup issue for this?

Reply:
Sure, I'll raise an issue!

Reply:
issue raised: https://github.com/drand/drand/issues/1086